### PR TITLE
set oauth2 as default security scheme and fix import command

### PR DESCRIPTION
**Pull request by VirajSalaka**

### CODE REVIEW

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java [hunk 1]
```diff
@@ -302,7 +302,11 @@ public static void setAdditionalConfigsDevFirst(ExtendedAPI api) {
         api.setEndpointConfigRepresentation(mgwEndpointConfigDTO);
         // 0th element represents the specific basepath
         api.setSpecificBasepath(basePath);
-        api.setApiSecurity(MgwDefinitionUtils.getSecurity(basePath));
+        String security = MgwDefinitionUtils.getSecurity(basePath);
+        if(security == null){
```
Comment (ruks) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java:306]: formatting issue

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java [hunk 2]
```diff
@@ -78,7 +78,17 @@ public void execute() {
...
 
-        MgwDefinitionUtils.setMgwDefinition(projectLocation + "/definition.yaml");
+        File importedAPIDefLocation = new File(GatewayCmdUtils.getProjectAPIDefinitionsDirectoryPath(projectName));
+        File addedAPIDefLocation = new File(GatewayCmdUtils.getProjectAPIFilesDirectoryPath(projectName));
+
+        if(importedAPIDefLocation.list().length == 0 && addedAPIDefLocation.list().length == 0 ){
+            throw new CLIInternalException("'api_definitions folder doesnot exist.");
+        }
+
+        if(importedAPIDefLocation.list().length > 0 && addedAPIDefLocation.list().length == 0 && !isCompiled){
```
Comment (ruks) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java:88]: formatting issue

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java [hunk 3]
```diff
@@ -78,7 +78,17 @@ public void execute() {
         projectName = projectName.replaceAll("[\\/\\\\]", "");
         File projectLocation = new File(GatewayCmdUtils.getProjectDirectoryPath(projectName));
 
-        MgwDefinitionUtils.setMgwDefinition(projectLocation + "/definition.yaml");
+        File importedAPIDefLocation = new File(GatewayCmdUtils.getProjectAPIDefinitionsDirectoryPath(projectName));
+        File addedAPIDefLocation = new File(GatewayCmdUtils.getProjectAPIFilesDirectoryPath(projectName));
+
+        if(importedAPIDefLocation.list().length == 0 && addedAPIDefLocation.list().length == 0 ){
```
Comment (ruks) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java:84]: formatting issue

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java [hunk 4]
```diff
@@ -78,7 +78,17 @@ public void execute() {
...
+            throw new CLIInternalException("'api_definitions folder doesnot exist.");
```
Comment (ruwiniwj) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java:85]: shall we change to `does not`

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java (hunk 1 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java:306]: code segment is no longer available.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java (hunk 2 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java:88]: already fixed via a later added PR.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java (hunk 3 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java:84]: already fixed via a later added PR.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java (hunk 4 above)
Comment (VirajSalaka) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java:85]: already fixed via a later added PR.

**Pull request merged**